scripts/tools/DeliveryExecutionReport.php

```diff
@@ -0,0 +1,197 @@
+
+    /** @var array */
+    private $params;
+
+    /** @var string */
+    private $method;
+
+    /**
+     * @inheritdoc
+     */
+    public function __invoke($params = [])
+    {
+        $this->init($params);
+        return call_user_func_array([$this, $this->method], $this->params);
+    }
+
+    /**
+     * @param array $params
+     * @throws
+     */
+    private function init($params = [])
+    {
+        if (!isset($params[0]) || !in_array($params[0], self::$validMethods)) {
+            throw new \InvalidArgumentException('Wrong method parameter. Available methods: ' . implode(',', self::$validMethods));
+        }
+        $this->method = $params[0];
+        $this->params = array_slice($params, 1);
+    }
+
+    /**
+     * Show comprehensive information about delivery execution
+     *
+     * Run example:
+     * ```
+     * sudo php index.php 'oat\taoSystemStatus\scripts\tools\DeliveryExecutionReport' show http://act-pr.docker.localhost/tao.rdf#i5e28154860c9a17943be5add4f7b1b5
+     * ```
+     *
+     * @param string $executionId
+     * @return Report
+     * @throws \common_exception_Error
+     * @throws \common_exception_NotFound
+     */
+    private function show($executionId)
+    {
+        $extensionManager = $this->getExtensionManager();
+        $deliveryExecutionService = $this->getDeliveryExecutionService();
+        $deliveryExecution = $deliveryExecutionService->getDeliveryExecution($executionId);
+        $report = new Report(Report::TYPE_INFO, sprintf('Information about delivery execution %s', $executionId));
+
+        $report->add(
+            new Report(
+                Report::TYPE_INFO,
+                'Delivery Id: ' . $deliveryExecution->getDelivery()->getUri() . PHP_EOL .
+                '  User Id: ' . $deliveryExecution->getUserIdentifier() . PHP_EOL .
+                '  Status: ' . $deliveryExecution->getState()->getLabel() . PHP_EOL .
+                '  Start Time: ' . $this->getFormatDate($deliveryExecution->getStartTime()) . PHP_EOL .
+                '  Finish Time: ' . $this->getFormatDate($deliveryExecution->getFinishTime()) . PHP_EOL .
+                '  Label: ' . $deliveryExecution->getLabel() . PHP_EOL
+            )
+        );
+
+        if ($extensionManager->isInstalled('taoProctoring')) {
+            $deliveryLog = $this->getDeliveryLog();
+            $deliveryLogEvents = $deliveryLog->search([DeliveryLog::DELIVERY_EXECUTION_ID => $executionId], [
+                'order' => DeliveryLog::CREATED_AT,
+                'dir' => 'asc',
+            ]);
+            $report->setData($deliveryLogEvents);
+            $monitoringService = $this->getDeliveryMonitoringService();
+            $deliveryMonitoringData = $monitoringService->getData($deliveryExecution)->get();
+            $report->setData($deliveryMonitoringData);
+        }
+
+        if ($extensionManager->isInstalled('taoAct')) {
+            $odsTaoActDeliveryLogCacheService = $this->getDeliveryLogCacheService();
```

Comment:
DeliveryLogCacheService is just a wrapper for DeliveryLog.
To get ODS events you need to use `\oat\taoAct\model\transmissionLog\TransmissionLogService`